docs/dependencies.md

```diff
@@ -1,80 +1,196 @@
 
-Building the firmware locally requires these dependencies ot be installed:
+There are two options for building the Particle firmware.
 
-1. [GCC for ARM Cortex processors](#1-gcc-for-arm-cortex-processors)
-2. [Make](#2-make)
-3. [Device Firmware Upgrade Utilities](#3-device-firmware-upgrade-utilities)
-4. [Zatig](#4-zatig) (for Windows users only)
-5. [Git](#5-git)
-6. [Command line tools](#6-command-line-tools)
+1. You can download and install the dependencies natively
+1. You can install Docker and run the toolchain in a container _(with all dependencies supplied therein)_
+
+Both options are presented below, and only one is required to build the Particle firmware.
+
+## Docker build container
+
+The Docker build container will execute `make`, and any parameters you pass behind it will be passed directly to `make`. Although more verbose, the Docker build container has a couple of advantages over native installation.
+
+1. Isolation of build dependencies required by the Particle firmware
+1. Dependencies are always in sync and are managed by Particle
+
+### Building Particle firmware
+
+```bash
+docker run --rm --volume <local firmware repository>:/particle-iot/firmware --workdir /particle-iot/firmware/modules particle/toolchain clean all PLATFORM=(core|electron|p1|photon)
+```
+
+**Option breakdown:**
+
+- `<local firmware repository>` - This is the path to the firmware folder on your local machine.
+- `(core|electron|p1|photon)` - Select the option from this list that matches the device for which you are compiling.
+
+### Building a user application
+
+```bash
+docker run --rm --volume <local firmware repository>:/particle-iot/firmware --volume <user application folder>:/particle-iot/app --workdir /particle-iot/firmware/main particle/toolchain clean all PLATFORM=(core|electron|p1|photon) APPDIR=/particle-iot/app TARGET_FILE=<user application>
+```
+
+**Option breakdown:**
+
+- `<local firmware repository>` - This is the path to the firmware folder on your local machine.
+- `<user application folder>` - This is the path to the user application folder on your local machine.
+- `(core|electron|p1|photon)` - Select the option from this list that matches the device for which you are compiling.
+- `<user application>` - This will be the name of the resulting binary (i.e. `my_app.elf`).
+
+> _**NOTE:** Special thanks to Kevin Sidwar for pioneering containerized builds in [his blog](https://www.kevinsidwar.com/iot/2018/3/1/local-particle-build-and-debug-through-docker-on-mac)._
+
+## Native Dependency Installation
+
+### Dependency List
+
+- [`arm-none-eabi-gcc`](#gcc-for-arm-cortex-processors)
+- [`make`](#gnu-make)
+- [`crc32`](#32-bit-cyclic-redundency-check-utility)
+- [`xxd`](#xxd-hex-dump-utility)
+- [Zadig (Windows only)](#zadig-windows-only)
+
+### GCC for ARM Cortex processors
+
+| Confirmed version | Download page |
+|:-----------------:|:-------------:|
+| 5.3.1 | [launchpad.net](https://launchpad.net/gcc-arm-embedded) |
```

Comment:
AFAIK ARM toolchains can no longer be downloaded from Launchpad. This link should instead be https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads